extensions/cl_ext_buffer_device_address.asciidoc

```diff
@@ -0,0 +1,320 @@
+:language: c
+
+// This is what is needed for C++, since docbook uses c++
+// and everything else uses cpp.  This doesn't work when
+// source blocks are in table cells, though, so don't use
+// C++ unless it is required.
+//:language: {basebackend@docbook:c++:cpp}
+
+== Name Strings
+
+`cl_ext_buffer_device_address`
+
+== Contact
+
+Pekka Jääskeläinen, Intel (pekka 'dot' jaaskelainen 'at' intel 'dot' com)
+
+== Contributors
+
+// spell-checker: disable
+Pekka Jääskeläinen, Intel +
+Karol Herbst, Red Hat +
+Henry Linjamäki, Intel +
+// spell-checker: enable
+
+== Notice
+
+Copyright (c) 2024 Intel Corporation.  All rights reserved.
+
+== Status
+
+Draft.
+
+== Version
+
+Built On: {docdate} +
+Revision: 0.3.0
+
+== Dependencies
+
+This extension is written against the OpenCL Specification version 3.0.16.
+
+This extension requires OpenCL 1.0 or later.
+
+== Overview
+
+The basic cl_mem buffer API doesn't enable access to the underlying raw
+pointers in the device memory, preventing its use in host side
+data structures that need pointer references to objects.
+
+Shared Virtual Memory (SVM) introduced in OpenCL 2.0 is the first feature
+that enables raw device side pointers in the OpenCL standard. Its coarse-grain
+variant is relatively simple to implement on various platforms in terms of
+coherency requirements, but it requires mapping the buffer's address range
+to the host virtual address space although it might not be needed by the
+application. This is not an issue in systems which can provide virtual memory
+across the platform, but might provide implementation challenges in cases
+where the device presents a global memory with a disjoint address space
+(that can also be a physical memory address space) or, for example, when
+a barebone embedded system lacks virtual memory support altogether.
+
+Various higher-level APIs present a memory allocation routine which can
+allocate device-only memory and provide raw pointers to it without guarentees
+of system-wide uniqueness: For example, minimal implementations of OpenMP's
+omp_target_alloc() and CUDA/HIP's cudaMalloc()/hipMalloc() do not require a shared
+address space between the host and the device. This extension is meant to
+provide a minimal set of features to implement such APIs using the cl_mem
+buffers without requiring a shared virtual address space between the host and
+the device.
+
+=== New API Function
+
+include::{generated}/api/protos/clSetKernelArgDevicePointerEXT.txt[]
+
+=== New API Enums
+
+Enums for enabling device pointer properties when creating a buffer
+{clCreateBuffer}, see <<clCreateBuffer, the list of supported memory flag values table>>:
+
+[source]
+----
+#define CL_MEM_DEVICE_SHARED_ADDRESS_EXT (1ul << 31)
+#define CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT (1ul << 30)
+----
+
+Enums for querying the device pointer from the cl_mem <<clGetMemObjectInfo, the list of supported param_names table>>:
+
+[source]
+----
+#define CL_MEM_DEVICE_PTR_EXT 0xff01
```

Comment:
This value does not match that defined in the XML. `CL_MEM_DEVICE_PTRS_EXT` should be mentioned here too.